packages/sync-server/src/sync/CentralSyncManager.ts

```diff
@@ -437,16 +450,121 @@ export class CentralSyncManager {
     }
   }
 
+  async persistIncomingChanges(sessionId: string, deviceId: string) {
+    const totalPushed = await countSyncSnapshotRecords(
+      this.models.database,
+      sessionId,
+      SYNC_SESSION_DIRECTION.INCOMING,
+    );
+    await this.models.syncSession.addInfo(sessionId, {
+      beganPersistAt: new Date(),
+      totalPushed,
+    });
+
+    try {
+      // commit the changes to the db
+      const persistedAtSyncTick = await this.models.wrapInTransaction(
+        async (transactingModels: SyncServerModelRegistry) => {
+          const modelsToInclude = getModelsForPush(transactingModels.getModels());
+
+          // we tick-tock the global clock to make sure there is a unique tick for these changes
+          // n.b. this used to also be used for concurrency control, but that is now handled by
```

Comment:
I'm not sure about whether we should include this comment or not, given it's never been used for concurrency control in Tupaia 🤔